usd/src/usd_parser/USDWorld.cc

```diff
@@ -302,6 +303,8 @@ namespace usd
         }
         sdf::usd::ParseUSDLinks(
           prim, linkName, optionalLink, usdData, scale->second);
+        *linkInserted = optionalLink.value();
+        linkScaleMap[linkName] = scale->second;
```

Comment:
I'm not sure that this line is needed. [A few lines above](https://github.com/ignitionrobotics/sdformat/blob/c3d53fa96bea6d6b530c8bd9f2d8c58b113639f3/usd/src/usd_parser/USDWorld.cc#L298-L303), we get the iterator to the pairing of the link name and scale, and then when `sdf::usd::ParseUSDLinks` is called before this assignment, the scale is passed as a [mutable reference](https://github.com/ignitionrobotics/sdformat/blob/c3d53fa96bea6d6b530c8bd9f2d8c58b113639f3/usd/src/usd_parser/USDLinks.hh#L57). So, I believe the assignment being done here (i.e., updating the link's scale) should be done automatically since we are using iterators and references. Do things still work if this line is removed?